postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -361,7 +371,18 @@ def _run_explain_safe(self, dbname, statement, obfuscated_statement):
                 tags=self._dbtags(dbname, "error:explain-{}".format(type(e))) + self._check._get_debug_tags(),
                 hostname=self._check.resolved_hostname,
             )
-            return None, DBExplainError.database_error, '{}'.format(type(err))
+            error_response = None, DBExplainError.database_error, '{}'.format(type(err))
+
+            if isinstance(e, psycopg2.errors.ProgrammingError) and not isinstance(
+                e, psycopg2.errors.InsufficientPrivilege
```

Comment:
In theory `InvalidSchemaName` can also be self-resolved.

Reply:
If it's an invalid schema for the explain_statement function then it would be caught earlier before this when we do `get_db_explain_setup_state` and we won't even try to explain anything until that's fixed.

Reply:
Ah good call. One more though: I ran into this error earlier on cloudsql hosts:

```
cannot collect execution plans due to failed DB connection to dbname=cloudsqladmin: OperationalError('FATAL:  pg_hba.conf rejects connection for host "[redacted]", user "datadog", database "cloudsqladmin", SSL off\n')
```

Reply:
This error is also caught earlier when we do `get_db_explain_setup_state`.